ir.ts

```diff
@@ -18,22 +18,25 @@ export type Stmt<A> =
   | {  a?: A, tag: "pass" }
   | {  a?: A, tag: "ifjmp", cond: Value<A>, thn: string, els: string }
   | {  a?: A, tag: "jmp", lbl: string }
-
   | { a?: A, tag: "store", start: Value<A>, offset: Value<A>, value: Value<A> } // start should be an id
-
+  | { a?: A, tag: "store_str", start: Value<A>, offset: Value<A>, value: Expr<A> } // start should be an id
+  | { a?: A, tag: "duplicate_str", source: Value<A>, dest: Value<A>}//duplicate the source string into the destination address
+  
 export type Expr<A> =
   | {  a?: A, tag: "value", value: Value<A> }
   | {  a?: A, tag: "binop", op: BinOp, left: Value<A>, right: Value<A>}
+  | {  a?: A, tag: "str_compare", op: Value<A>, left:Value<A>, right: Value<A> } // op: 0 for inequal and 1 for equal
```

Comment:
Why can't these all be implemented as built-in functions?

Reply:
Yeah. Actually, it is a built-in function implemented using wasm, we can do some reformatting later.

Reply:
I'm trying to merge this, and can't really figure out what to do. It looks like `str_compare` got moved to a new IR form... but I don't understand that either. The IR form has new stuff in the compiler that refers to a built-in `str_comparison` (great), but then does extra comparisons and has another special `op` field.

Why can't the design be that string comparisons are just a kind of binop that lowers to a call to a WASM function like `str_comparison` directly? Then we don't need an expr _or_ and IR form for it.

Reply:
Hi, we update the work in the pull request of week 8 which is #60.